tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -207,39 +244,111 @@ build_compressed_scan_pathkeys(SortInfo *sort_info, PlannerInfo *root, List *chu
 	 */
 	if (sort_info->needs_sequence_num)
 	{
-		bool nulls_first;
-		Oid sortop;
-		varattno =
-			get_attnum(info->compressed_rte->relid, COMPRESSION_COLUMN_METADATA_SEQUENCE_NUM_NAME);
-		var = makeVar(info->compressed_rel->relid, varattno, INT4OID, -1, InvalidOid, 0);
-
-		if (sort_info->reverse)
+		if (info->has_seq_num)
```

Comment:
Nit: I would suggest to break out these two paths in two separate functions, one for legacy seqnos and one for no seqnos. Can also be refactored later.

Reply:
Will add a TODO to cleanup in a followup.